modules/pam_env/pam_env.c

```diff
@@ -682,19 +886,36 @@ static int
 _parse_env_file(pam_handle_t *pamh, int ctrl, const char *file)
 {
     int retval=PAM_SUCCESS, i, t;
-    char buffer[BUF_SIZE], *key, *mark;
-    FILE *conf;
+    char *key, *mark;
+    char **env_list = NULL;
+
+#ifdef VENDOR_DEFAULT_ETC_ENVFILE
+#ifndef USE_ECONF
+    /* Only one file will be parsed. So, file has to be set. */
+    if (file == NULL) /* No filename has been set via argv. */
+      file = DEFAULT_ETC_ENVFILE;;
+    /*
+    * Check whether DEFAULT_ETC_ENVFILE; file is available.
+    * If it does not exist, fall back to VENDOR_DEFAULT_ETC_ENVFILE; file.
+    */
+    struct stat stat_buffer;
+    if (stat(file, &stat_buffer) != 0 && errno == ENOENT) {
+      file = VENDOR_DEFAULT_ETC_ENVFILE;
+    }
+#endif
+#endif
```

Comment:
In `USE_ECONF` case this means that `file` can be `NULL`, and this is problematic because `file` is used later in this function for error diagnostics.

Reply:
I am ensuring now that file is not NULL while reading the file via the "old style".